src/Check/SecurityAdvisory.php

```diff
@@ -56,14 +56,16 @@ public function __construct($lockFilePath = null)
         }
 
         $this->lockFilePath    = $lockFilePath;
-
-        $parser = new AdvisoryParser((new AdvisoryFetcher)->fetchAdvisories());
-
-        $this->advisoryAnalyzer = new AdvisoryAnalyzer($parser->getAdvisories());
     }
 
     public function check()
     {
+        if ($this->advisoryAnalyzer === null) {
```

Comment:
Is there any meaningful way to test this behavior, perhaps? Maybe instantiate the `SecurityAdvisory` class with a `$lockFilePath` that contains invalid JSON?

Reply:
Okay, I'm a bit lost, to be honest. I kept the AdvisoryAnalyzer as a class property to make sure the current tests keep running.

Reply:
#28 reports a **problem**.

What we need to do, to prevent future regressions, is to reproduce said problem, and verify that it doesn't occur.

In practice, in the test:

```
$check = new SecurityAdvisory(...);

$this->expectException(...);

$advisory->check();
```

Reply:
Okay, but in the end that's impossible without refactoring the while SecurityAdvisory class and wrapping some dependency injection magic around it, isn't it? What needs to be proven/tested is that there is no http call in the SecurityAdvisory constructor and it is outside of my knowledge on how to prove the absence of that behaviour. 
  
I did add a test for the exception we saw in production - not sure how helpful it is. 

In the end the transfer of the `fetchAdvisories` function from the constructor into the `check` function is just fighting symptoms anyway.
The real problem happens here: https://github.com/enlightn/security-checker/blob/main/src/Filesystem.php#L21 where somehow between line 21 and 25 the directory vanishes. ("Somehow" in this case was because multiple instances of the SecurityAdvisory class were being created who all wrote to and deleted from the temporary directory. And the only way to solve _that_ problem is probably some kind of locking mechanism on the OS level.

Reply:
> Okay, but in the end that's impossible without refactoring the while SecurityAdvisory class and wrapping some dependency injection magic around it, isn't it?

No, if you create a `new SecurityAdvisory()` with an invalid lock file, it should crash as per `1.11.x`.

With your patch, it should crash when you call `->check()`.

Reply:
So, once again I'm a bit stumped. As far as I can tell it now works as intended, but I'm not sure how to test the whole thing. Couldn't really think of a good solution in the last week, either.

Reply:
The only thing I'd say we should get rid of, before merging, is to get rid of `$this->advisoryFetcher`, and instead always instantiate a new one in the `->check()` method.

Reply:
Yeah, you're absolutely right. I added this for a test that basically doesn't really test anything worth testing. I'm removing the AdvisoryFetcher and the test that used it.

Reply:
So, here go.

Reply:
I still see the property?

Reply:
That's the analyzer, not the fetcher. There are a bunch of tests using the analyzer.

Reply:
Yes, and now it's nullable - shouldn't the property just go away?

Reply:
If you have a good idea how to run these tests: https://github.com/laminas/laminas-diagnostics/blob/1.10.x/test/ChecksTest.php#L558 without the property, I'd be happy to remove it, too.

Reply:
I've re-read all my comments: gonna dismiss them as my own brainfart, sorry for all the noise 😬